experimental/segmenter/src/word.rs

```diff
@@ -15,17 +16,57 @@ use icu_provider::prelude::*;
 use utf8_iter::Utf8CharIndices;
 
 /// Word break iterator for an `str` (a UTF-8 string).
-pub type WordBreakIteratorUtf8<'l, 's> = RuleBreakIterator<'l, 's, WordBreakTypeUtf8>;
+///
+/// For more information, see [`WordSegmenter`].
+#[derive(Debug)]
+pub struct WordBreakIteratorUtf8<'l, 's>(RuleBreakIterator<'l, 's, WordBreakTypeUtf8>);
```

Comment:
You should move `complex` out of `RuleBreakIterator` and make it a non-optional field here and in `LineBreakIterator`. It's required for word and line, and not used for the other segmenters.

Reply:
I did it for `LineBreakIterator` which was fairly trivial. It's a bit more involved for `RuleBreakIterator` and I ran out of time for now.